pkg/cmd/util/clientcmd/factory.go

```diff
@@ -169,6 +171,39 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 			return nil, fmt.Errorf("it is not possible to get ports from %s", kind)
 		}
 	}
+	w.LogsForObject = func(object runtime.Object, container string, follow, previous bool) (io.ReadCloser, error) {
+		client, err := clients.ClientForVersion("")
```

Comment:
Empty string here looks a little weird.  Don't we normally pull this from the `RESTMapping`?